tests/fuser_mqubit_test.cc

```diff
@@ -644,6 +648,333 @@ TEST(FuserMultiQubitTest, SmallCircuits) {
+  }
+
+  {
+    unsigned num_qubits = 6;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(0, {2, 3}),
+      CreateDummyGate(1, {1, 2}),
+      CreateDummyGate(0, {4, 5}),
+      CreateDummyGate(1, {3, 4}),
+      CreateDummyMeasurementGate(2, {0, 1, 2}),
+      CreateDummyMeasurementGate(2, {4, 5}),
+      CreateDummyGate(3, {0, 1}),
+      CreateDummyGate(3, {2, 3}),
+      CreateDummyGate(4, {1, 2}),
+      CreateDummyGate(3, {4, 5}),
+      CreateDummyGate(4, {3, 4}),
+    };
+
+    param.max_fused_size = 6;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_EQ(fused_gates.size(), 3);
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+  }
+
+  {
+    unsigned num_qubits = 6;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(0, {2, 3}),
+      CreateDummyGate(1, {1, 2}),
+      CreateDummyGate(0, {4, 5}),
+      CreateDummyGate(1, {3, 4}),
+      CreateDummyGate(1, {5, 0}),
+    };
+
+    param.max_fused_size = 3;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_EQ(fused_gates.size(), 4);
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+  }
+
+  {
+    unsigned num_qubits = 8;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(0, {2, 3}),
+      CreateDummyGate(0, {4, 5}),
+      CreateDummyGate(1, {1, 2}),
+      CreateDummyGate(1, {3, 4}),
+      CreateDummyGate(0, {6, 7}),
+      CreateDummyGate(1, {5, 6}),
+      CreateDummyGate(1, {7, 0}),
+    };
+
+    param.max_fused_size = 5;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_EQ(fused_gates.size(), 3);
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+  }
+
+  {
+    unsigned num_qubits = 6;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(1, {2, 3}),
+      CreateDummyGate(3, {1, 2}),
+      CreateDummyGate(2, {4, 5}),
+      CreateDummyGate(4, {2, 3, 4}),
+    };
+
+    param.max_fused_size = 3;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_EQ(fused_gates.size(), 4);
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+  }
+
+  {
+    unsigned num_qubits = 7;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {1, 6}),
+      CreateDummyGate(1, {0, 1}),
+      CreateDummyGate(2, {1, 2, 3}),
+      CreateDummyGate(1, {4, 5}),
+      CreateDummyGate(3, {3, 4}),
+      CreateDummyGate(2, {5, 6}),
+    };
+
+    {
+      param.max_fused_size = 3;
+      auto fused_gates = Fuser::FuseGates(
+          param, num_qubits, circuit.begin(), circuit.end());
+
+      EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
```

Comment:
Should this test include a `fused_gates.size()` check? Same for the one immediately below this.

Reply:
Done.